airflow/cli/commands/config_command.py

```diff
@@ -17,11 +17,54 @@
 """Config sub-commands"""
 import io
 
-from airflow.configuration import conf
+from configupdater import ConfigUpdater, NoSectionError
+
+from airflow.configuration import AIRFLOW_CONFIG, AirflowConfigException, conf
 
 
 def show_config(args):
     """Show current application configuration"""
     with io.StringIO() as output:
-        conf.write(output)
+        if args.section:
+            section = args.section
+            if conf.has_section(section):
+                conf._write_section(  # pylint: disable=protected-access
+                    fp=output,
+                    section_name=section,
+                    section_items=conf.getsection(section).items(),
+                    delimiter=" = "
+                )
+            else:
+                raise AirflowConfigException(f"No section '{section}'")
+        else:
+            conf.write(output)
```

Comment:
Let's add a confirmation prompt with the option and value

Reply:
I don't remember cany CLI to ask me if I am sure about changing config. In case Airflow there's one big problem: if you set executor to something like `NonExisting` then there will be no option to fix it using CLI because everything is loaded before...

Reply:
This is to safeguard against accidental rewrite of fernet key, sql_alchemy_conn.

Ansible, Terraform have their prompts on different command.

It is similar to how not many Apps allow changing config via CLI